File: patch_backend.py
with open("admin_backend.js", "r", encoding="utf-8") as f:
    text = f.read()

# Edit /metricas to include top_faltosos
old_metricas = """const team = await pool.query("SELECT COUNT(*) FROM treinadores");
        
        res.json({ 
            success: true, 
            total_atletas: tot.rows[0].count,
            departamento_medico: dm.rows[0].count,
            equipe_tecnica: team.rows[0].count
        });"""

new_metricas = """const team = await pool.query("SELECT COUNT(*) FROM treinadores");
        
        let faltasQ = "SELECT a.nome, COUNT(p.id) as faltas FROM presencas p JOIN atletas a ON p.atleta_id = a.id WHERE p.status = 'F' GROUP BY a.id ORDER BY faltas DESC LIMIT 5";
        if (req.usuario.perfil === 'Treinador') {
            faltasQ = "SELECT a.nome, COUNT(p.id) as faltas FROM presencas p JOIN atletas a ON p.atleta_id = a.id JOIN treinador_categoria tc ON a.categoria_id = tc.categoria_id WHERE tc.treinador_id = $1 AND p.status = 'F' GROUP BY a.id ORDER BY faltas DESC LIMIT 5";
        }
        const topFaltosos = await pool.query(faltasQ, vals);
        
        res.json({ 
            success: true, 
            total_atletas: tot.rows[0].count,
            departamento_medico: dm.rows[0].count,
            equipe_tecnica: team.rows[0].count,
            top_faltosos: topFaltosos.rows
        });"""

text = text.replace(old_metricas, new_metricas)

# Add /eventos/proximos
new_route = """// ==========================================
// PRÓXIMOS JOGOS
// ==========================================
router.get('/eventos/proximos', verificarAcesso, async (req, res) => {
    try {
        let q = "SELECT t.id, TO_CHAR(t.data, 'DD/MM/YYYY') as data_br, t.titulo, c.nome as categoria_nome FROM treinos t LEFT JOIN categorias c ON t.categoria_id = c.id WHERE t.tipo = 'JOGO' AND t.data >= CURRENT_DATE AND t.data <= CURRENT_DATE + INTERVAL '10 days' ORDER BY t.data ASC";
        let vals = [];
        
        if (req.usuario.perfil === 'Treinador') {
            q = "SELECT t.id, TO_CHAR(t.data, 'DD/MM/YYYY') as data_br, t.titulo, c.nome as categoria_nome FROM treinos t LEFT JOIN categorias c ON t.categoria_id = c.id JOIN treinador_categoria tc ON t.categoria_id = tc.categoria_id WHERE tc.treinador_id = $1 AND t.tipo = 'JOGO' AND t.data >= CURRENT_DATE AND t.data <= CURRENT_DATE + INTERVAL '10 days' ORDER BY t.data ASC";
            vals.push(req.usuario.id);
        }
        
        const r = await pool.query(q, vals);
        res.json(r.rows);
    } catch (e) {
        res.status(500).json({ error: e.message });
    }
});

module.exports = router;"""

text = text.replace("module.exports = router;", new_route)

with open("admin_backend.js", "w", encoding="utf-8") as f:
    f.write(text)

File: patch_backend.js
const fs = require('fs');
let code = fs.readFileSync('admin.js', 'utf8');

// 1. Update GET /atletas
const oldGet = `        let query = \`
            SELECT a.*, c.nome as categoria 
            FROM atletas a 
            LEFT JOIN categorias c ON a.categoria_id = c.id 
        \`;
        let values = [];

        if (req.usuario.perfil === 'Treinador') {
            query += \`
                JOIN treinador_categoria tc ON tc.categoria_id = a.categoria_id
                WHERE tc.treinador_id = $1
            \`;
            values.push(req.usuario.id);
        }`;

const newGet = `        let query = \`
            SELECT a.*, c.nome as categoria 
            FROM atletas a 
            LEFT JOIN categorias c ON a.categoria_id = c.id 
            WHERE a.ativo = true
        \`;
        let values = [];

        if (req.usuario.perfil === 'Treinador') {
            query += \`
                AND a.categoria_id IN (SELECT categoria_id FROM treinador_categoria WHERE treinador_id = $1)
            \`;
            values.push(req.usuario.id);
        }`;

if(code.includes(oldGet)) {
    code = code.replace(oldGet, newGet);
    console.log("Patched GET /atletas");
} else {
    console.log("Could not find GET /atletas block");
}

// 2. Add GET /atletas/inativos BEFORE GET /atletas/:id
const inativosEndpoint = `
router.get('/atletas/inativos', verificarAcesso, async (req, res) => {
    try {
        let query = \`
            SELECT a.*, c.nome as categoria 
            FROM atletas a 
            LEFT JOIN categorias c ON a.categoria_id = c.id 
            WHERE a.ativo = false 
        \`;
        let values = [];

        if (req.usuario.perfil === 'Treinador') {
            query += \`
                AND a.categoria_id IN (SELECT categoria_id FROM treinador_categoria WHERE treinador_id = $1)
            \`;
            values.push(req.usuario.id);
        }
        
        query += " ORDER BY a.nome";
        const result = await pool.query(query, values);
        res.json({ success: true, atletas: result.rows });
    } catch (error) { console.error(error); res.status(500).json({ error: 'Erro interno' }); }
});
`;

if (!code.includes('/atletas/inativos')) {
    code = code.replace("router.get('/atletas/:id'", inativosEndpoint + "\nrouter.get('/atletas/:id'");
    console.log("Added GET /atletas/inativos");
}

// 3. Update DELETE /atletas/:id
const oldDelete = `await pool.query("DELETE FROM atletas WHERE id = $1", [req.params.id]);`;
const newDelete = `await pool.query("UPDATE atletas SET ativo = false WHERE id = $1", [req.params.id]);`;
if(code.includes(oldDelete)) {
    code = code.replace(oldDelete, newDelete);
    console.log("Patched DELETE /atletas/:id");
} else {
    console.log("Could not find DELETE block");
}

fs.writeFileSync('admin.js', code, 'utf8');
